app/src/hooks/useCounter.tsx

```diff
@@ -0,0 +1,46 @@
+import { useState, useEffect } from 'react'
+
+const DEFAULT_VALUES = {
+  COUNTER_START: 0,
+  COUNTING_INTERVAL: 1,
+} as const
+
+const calculateCounter = (prevCounter: number, seconds: number, isIncrementing: boolean) => {
+  if (isIncrementing) {
+    return prevCounter < seconds ? prevCounter + DEFAULT_VALUES.COUNTING_INTERVAL : seconds
+  } else {
+    return prevCounter > 0 ? prevCounter - DEFAULT_VALUES.COUNTING_INTERVAL : 0
+  }
+}
+
+export default function useGetCounter(seconds: number, isCountingUp: boolean) {
+  const [counterContent, setCounterContent] = useState(
+    isCountingUp ? DEFAULT_VALUES.COUNTER_START : seconds,
+  )
+  const [isPaused, setIsPaused] = useState(false)
+  const [isFinished, setIsFinished] = useState(false)
+
+  const handlePause = () => {
+    setIsPaused((isPaused) => !isPaused)
+  }
+
+  useEffect(() => {
+    const timer = setInterval(() => {
+      if (isPaused || isFinished) return
+
+      setCounterContent((prevCounter) => {
+        return calculateCounter(prevCounter, seconds, isCountingUp)
+      })
+    }, DEFAULT_VALUES.COUNTING_INTERVAL * 1000)
+
+    return () => clearInterval(timer)
+  }, [isCountingUp, seconds, isPaused])
+
+  useEffect(() => {
+    if ((!isCountingUp && counterContent === 0) || (isCountingUp && counterContent === seconds)) {
+      setIsFinished(true)
+    }
+  }, [counterContent])
+
+  return { counterContent, handlePause, isPaused, isFinished }
```

Comment:
`counterContent` should be the time formatted (for example `12:30`).

`counterValue` should be the time in seconds.